### PR TITLE
TMDM-14812 Delete a FK points to inheritance type raise an exception

### PR DESCRIPTION
**Pull request by hw-zhu**
https://jira.talendforge.org/browse/TMDM-14812
**What is the current behavior?** (You should also link to an open issue here)
Using attached DM to generate physical table that having a FK constraint points to referenced table being inheritance type, whereas, some invalid foreign key name generated when deploy the dm, the exception occur when drop the invalid fk column as the expected results.  

**What is the new behavior?**
As the solution, when prepare to drop the FK constraint, need to further check the existing of foreign key name in the database, if not exist, skip to delete the special FK.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by zbcui — APPROVED** [review]

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/mapping/MDMTable.java [hunk 1]
```diff
@@ -23,8 +22,8 @@
 import java.util.Properties;
 
 import org.apache.commons.lang3.StringUtils;
-import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/mapping/MDMTable.java:26]: Does this file need change?

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/AbstractLiquibaseSchemaAdapter.java [hunk 2]
```diff
@@ -44,12 +53,12 @@
 
     public static final String MDM_ROOT = "mdm.root"; //$NON-NLS-1$
 
-    protected RDBMSDataSource dataSource;
+    protected static RDBMSDataSource dataSource;
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/AbstractLiquibaseSchemaAdapter.java:56]: Can we know the reason to change this?

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/AbstractLiquibaseSchemaAdapter.java [hunk 3]
```diff
@@ -136,4 +145,152 @@ protected String generateChangeLogFile(liquibase.changelog.DatabaseChangeLog dat
...
+            List<String> parameters = new ArrayList<>();
+            parameters.add(curFKCheckingSource.getQualityName(tableName));
+            parameters.add(curFKCheckingSource.getQualityName(constraintName));
+            result = executeSQL(sql, parameters);
+            if (Integer.parseInt(result) > 0) {
+                LOGGER.info("Query retrieves foreign key constraints on the given table "+ tableName + " with constraint name " + constraintName);
+                return true;
+            }
+        } catch (Exception e) {
+            LOGGER.error("Failed to execute sql statement, \n caused by ", e);
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/AbstractLiquibaseSchemaAdapter.java:173]: Why needs a `\\n` in log?

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/AbstractLiquibaseSchemaAdapter.java (hunk 2 above)
Comment (hw-zhu) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/AbstractLiquibaseSchemaAdapter.java:56]: want to use it in static enum class, doesn't affect function.

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/AbstractLiquibaseSchemaAdapter.java (hunk 3 above)
Comment (hw-zhu) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/AbstractLiquibaseSchemaAdapter.java:173]: Hope to wrap a line and get error information like the following
```
Caused by: XXXException: reason
```

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/mapping/MDMTable.java (hunk 1 above)
Comment (hw-zhu) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/mapping/MDMTable.java:26]: Format code in eclipse to auto move the line.

**Review by zbcui — APPROVED** [review]

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/AbstractLiquibaseSchemaAdapter.java (hunk 3 above)
Comment (VictorLi-talendbj) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/AbstractLiquibaseSchemaAdapter.java:173]: Can this be changed to:
`LOGGER.error("Failed to execute SQL statement.", e);`

Comment (VictorLi-talendbj) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/AbstractLiquibaseSchemaAdapter.java:173]: Ok

**Review by VictorLi-talendbj — APPROVED** [review]

**Pull request merged**